### ISSUE TITLE
Const for create process of Linear solver factory

### ISSUE THREAD

**Issue by adityaghantasala** [KratosMultiphysics/Kratos#5193]
I am trying to use the linear solver factory in the following way 

```cpp
LinearSolverFactoryType const& linear_solver_factory = KratosComponents<LinearSolverFactoryType>::Get("amgcl");
auto amgcl_solver = linear_solver_factory.Create(amgcl_settings);
```

and am getting a compilation error saying no function ```..... Create(Parameters ) const```

Î think there is const missing for the Create method of the linear solver factory. 

https://github.com/KratosMultiphysics/Kratos/blob/ef386f8309094526e5aaf9cb8d096671b704d5a8/kratos/factories/linear_solver_factory.h#L101

It should be 

```cpp
virtual typename LinearSolver<TSparseSpace,TLocalSpace>::Pointer Create(Kratos::Parameters Settings) const
```

can some one conform this ?

**Comment by loumalouomega** [comment]
Well I agree, makes sense, the Factory only does one task

**Comment by loumalouomega** [comment]
> Well I agree, makes sense, the Factory only does one task

And doesn't require to modify values a posteriori

**Comment by adityaghantasala** [comment]
Also the check 

https://github.com/KratosMultiphysics/Kratos/blob/ef386f8309094526e5aaf9cb8d096671b704d5a8/kratos/factories/linear_solver_factory.h#L108

is redundant as the KratosComponents already does the check.

**Comment by loumalouomega** [comment]
> is redundant as the KratosComponents already does the check.

Huum, I am not sure, the Factory registers in a particular manner

**Comment by adityaghantasala** [comment]
That would require Has to also be a const ! But I think it has to be like that  or ?

**Comment by philbucher** [comment]
Solved in #5194
